### PR TITLE
Fix HTTP listen behavior being non-compliant with node

### LINKED ISSUE #4651: Fastify "Is port ... in use?"
### What version of Bun is running?

_No response_

### What platform is your computer?

_No response_

### What steps can reproduce the bug?

```
import { fastify } from 'fastify';

const server = fastify({ logger: true });
await server.listen({ port: 3000 });
```

### What is the expected behavior?

_No response_

### What do you see instead?

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by SuperAuguste**
### What does this PR do?

Closes #4651

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

- [x] I ran `make js` and committed the transpiled changes
- [x] I or my editor ran Prettier on the changed files (or I ran `bun fmt`)
- [x] I included a test for the new code, or an existing test covers it

### CODE REVIEW

**Review by paperclover — APPROVED** [review]
nice

**Comment by hyuzea** [comment]
will this emits `ADDRINUSE` error? so this also fixes #3767

**Comment by SuperAuguste** [comment]
@yus-ham do you have an exact repro I could use to test this? :)

**Comment by hyuzea** [comment]
> @yus-ham do you have an exact repro I could use to test this? :)

try this one
```js
try {
  createServer(() => {}).listen(7890)
  createServer(() => {}).listen(7890)
} catch(err) {
  expect(err.code).toBe('EADDRINUSE')
}
```

ps. vite does check for `ADDRINUSE`
https://github.com/vitejs/vite/blob/6ffe0703e2ba968705832dd58ab2ed78deee8d22/packages/vite/src/node/http.ts#L158-L165

**Comment by SuperAuguste** [comment]
Modified it to more closely match Vite:
```ts
import {createServer} from "http"
createServer(() => {}).listen(7890, () => {
    createServer(() => {}).on("error", err => console.log(err.code)).listen(7890)
})
```
and it does not return a code like Node; I'll fix this ASAP :)

EDIT: Looks like this should be handled in Bun.serve

**Comment by SuperAuguste** [comment]
Screenshot 2023-09-10 at 11 34 40 AM

This should do the trick!

File: src/bun.js/api/server.zig [hunk 1]
```diff
@@ -5393,6 +5393,7 @@ pub fn NewServer(comptime NamespaceType: type, comptime ssl_enabled_: bool, comp
                         ).toErrorInstance(
                             this.globalThis,
                         );
+                        error_instance.put(this.globalThis, ZigString.static("code"), ZigString.static("EADDRINUSE").toValueAuto(this.globalThis));
```
Comment (paperclover) [inline src/bun.js/api/server.zig]: we can use SystemError{...}.toErrorInstance

**Comment by paperclover** [comment]
oh god i rebased this stupidly

**Comment by paperclover** [comment]
see other pr, sorry

**Pull request closed without merging**